src/models/room.js

```diff
@@ -189,6 +191,16 @@ export function Room(roomId, client, myUserId, opts) {
 
     if (this._opts.pendingEventOrdering == "detached") {
         this._pendingEventList = [];
+        const serializedPendingEventList = client._sessionStore.store.getItem(pendingEventsKey(this.roomId));
+        if (serializedPendingEventList) {
+            JSON.parse(serializedPendingEventList)
+                .forEach(serializedEvent => {
+                    const event = new MatrixEvent(serializedEvent);
+                    event.setStatus(EventStatus.NOT_SENT);
+                    const txnId = client.makeTxnId();
```

Comment:
I think it would be good to have the txnId be persisted to ensure idempotency even if the user refreshed during a server timeout/whilst sending rather than generating a fresh one

Reply:
Addressed in matrix-org/matrix-js-sdk#663